### PR TITLE
Added scheduled cloud function module

### PR DESCRIPTION
**Pull request by cleentfaar**
Moved from ingestors to here for reusability

Also added a few more optional variables to allow for different runtimes and entry_points

### CODE REVIEW

File: terraform/cloud_function/scheduled/main.tf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  source_archive_object         = google_storage_bucket_object.functioncode.name
+  timeout                       = var.function_timeout
+  trigger_http                  = true
+  vpc_connector                 = var.vpc_connector
+  vpc_connector_egress_settings = var.vpc_connector_egress_settings
+}
+
+# Add cloud scheduler job
+resource "google_cloud_scheduler_job" "casco_listing_job" {
+  count    = var.branch_suffix == "" ? 1 : 0 # Only enable job on production to avoid branches eating each others lunch
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/main.tf:39]: Having this logic here is very ugly. We can make this a variable, `var.enable_scheduler` or something, and leave the logic of when to enable in the place where the module is used

File: terraform/cloud_function/scheduled/variables.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+variable "function_name" {}
+variable "function_runtime" {
+  default = "python39"
+}
+variable "function_service_account_email" {}
+variable "function_timeout" { default = 60 }
+variable "iam_invoke_member_email" {}
+variable "project_id" {}
+variable "project_region" {}
+variable "request_body" {
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/variables.tf:21]:
```suggestion
variable "scheduler_request_body" {
```

File: terraform/cloud_function/scheduled/variables.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  default = "{}"
+}
+variable "request_method" {
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/variables.tf:24]:
```suggestion
variable "scheduler_request_method" {
```

File: terraform/cloud_function/scheduled/variables.tf [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  default = "POST"
+}
+variable "retry_count" {
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/variables.tf:27]:
```suggestion
variable "scheduler_retry_count" {
```

File: terraform/cloud_function/scheduled/variables.tf [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  default = 1
+}
+variable "schedule" {
+  description = "See https://cloud.google.com/scheduler/docs/configuring/cron-job-schedules"
+}
+variable "source_code_bucket_name" {}
+variable "source_code_root_path" {}
+variable "vpc_connector" {
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/variables.tf:35]: Why does this function have this but not the pubsub one?

File: terraform/cloud_function/scheduled/variables.tf [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  default = {}
+}
+variable "function_memory" {}
+variable "function_name" {}
+variable "function_runtime" {
+  default = "python39"
+}
+variable "function_service_account_email" {}
+variable "function_timeout" { default = 60 }
+variable "iam_invoke_member_email" {}
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/variables.tf:18]:
```suggestion
variable "function_invoke_member" {}
```

File: terraform/cloud_function/scheduled/variables.tf [hunk 7]
```diff
@@ -0,0 +1,40 @@
+variable "attempt_deadline" {
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/variables.tf:1]:
```suggestion
variable "scheduler_attempt_deadline" {
```

**Review by barendregt — CHANGES_REQUESTED** [review]

File: terraform/cloud_function/scheduled/variables.tf (hunk 5 above)
Comment (cleentfaar) [reply terraform/cloud_function/scheduled/variables.tf:35]: no reason really, just that scheduled functions tend to be used for accessing external stuff (up til now), whilst pubsub ones are more event driven / internal

happy to add though

File: terraform/cloud_function/scheduled/main.tf (hunk 1 above)
Comment (cleentfaar) [reply terraform/cloud_function/scheduled/main.tf:39]: agree 👍

File: terraform/cloud_function/scheduled/variables.tf [hunk 8]
```diff
@@ -0,0 +1,43 @@
+variable "branch_suffix" {}
+variable "function_entry_point" {
+  default = "handler"
+}
+variable "function_env_vars" {
+  default = {}
+}
+variable "function_memory" {}
```
Comment (barendregt) [inline terraform/cloud_function/scheduled/variables.tf:8]: This should have a default value like `128`

**Review by barendregt — APPROVED** [review]

**Pull request merged**